### PR TITLE
run with a docker container

### PR DESCRIPTION
**Pull request by rberrelleza**
Initial version of the Dockerfile.  It's not optimized for production just yet, but wanted to send something that works and then build up from there. 

Long term, I think moving to using [phoenix releases](https://hexdocs.pm/phoenix/releases.html) makes sense, since it allows you to have a much smaller docker image.  Also, not sure what you are using for releases, but it would be cool to hook this up for that (we use github releases to ship containers with well-known versions, it works pretty well)

Signed-off-by: Ramiro Berrelleza <rberrelleza@gmail.com>

### CODE REVIEW

**Comment by rberrelleza** [comment]
I also build a helm chart (so I could  run this in kubernetes), let me know if you are interested and I can send a separate PR.

File: config/dev.exs [hunk 1]
```diff
@@ -1,11 +1,11 @@
 use Mix.Config
 
+database_url =
+  System.get_env("DATABASE_URL") || "ecto://postgres:postgres@localhost/chat_api_dev"
```
Comment (rberrelleza) [inline config/dev.exs:4]: I made this change so it was easier to test when running with non-local databases (e.g. in my case I run everything on a remote cluster, so i needed to change the hostname while still running in dev mode)

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+set -euo pipefail
+
+POOL_SIZE=2 mix ecto.setup
```
[the remark this answers is not in the document]
Comment (reichert621) [reply docker-entrypoint.sh:4]: i believe `ecto.setup` encapsulates both `ecto.create` and `ecto.migrate`, so i ~think it shouldn't be necessary

File: config/dev.exs (hunk 1 above)
Comment (reichert621) [reply config/dev.exs:4]: nice!

**Comment by reichert621** [comment]
thanks @rberrelleza! this is great 🎉 

cc @cheeseblubber mind taking this? you probably have more familiarity with this part of the code 😬

File: config/dev.exs [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+
 # Configure your database
 config :chat_api, ChatApi.Repo,
-  username: "postgres",
-  password: "postgres",
-  database: "chat_api_dev",
-  hostname: "localhost",
```
Comment (reichert621) [inline config/dev.exs:8]: @cheeseblubber will this break things for us?

Comment (cheeseblubber) [reply config/dev.exs:8]: This is fine because its specified in the database URL

**Comment by syabro** [comment]
@rberrelleza what's the difference between "dev" and "production" here?
I mean what's wrong if I delete node part from dockerfile and run the server?

**Comment by rberrelleza** [comment]
@syabro the 'dev' and 'prod' env var controls how assets are build. You can't remove the node part, because you need it to build the webpack assets. 

There's a pending follow up on this task to optimize the docker image to make it smaller, so that the final image just have the assets. But the Dockerfile, as it is today, doesn't support that.

**Comment by syabro** [comment]
@rberrelleza emm, actually I want to import react stuff in my main react application and use it there and keep the image clean

**Pull request merged**